lib/ansible/modules/cloud/amazon/cloudfront_facts.py

```diff
@@ -0,0 +1,453 @@
+            self.module.fail_json(msg="Error listing cloud front origin access identities = " + str(e), exception=traceback.format_exc(e))
+
+    def list_distributions(self):
+        try:
+            func = partial(self.client.list_distributions)
+            return self.paginated_response(func, 'DistributionList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing distributions = " + str(e), exception=traceback.format_exc(e))
+
+    def list_distributions_by_web_acl(self, web_acl_id):
+        try:
+            func = partial(self.client.list_distributions, web_acl_id)
+            return self.paginated_response(func, 'DistributionList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing distributions by web acl id = " + str(e), exception=traceback.format_exc(e))
+
+    def list_invalidations(self):
+        try:
+            func = partial(self.client.list_invalidations)
+            return self.paginated_response(func, 'InvalidationList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing invalidations = " + str(e), exception=traceback.format_exc(e))
+
+    def list_streaming_distributions(self):
+        try:
+            func = partial(self.client.list_streaming_distributions)
+            return self.paginated_response(func, 'StreamingDistributionList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing streaming distributions = " + str(e), exception=traceback.format_exc(e))
+
+    def get_distribution_id_from_domain_name(self, domain_name):
+        try:
+            distribution_id = ""
+            distributions = self.list_distributions()
+            for dist in distributions['Items']:
+                for alias in dist['Aliases']['Items']:
+                    if str(alias).lower() == domain_name.lower():
+                        distribution_id = str(dist['Id'])
+                        break
+            return distribution_id
+        except Exception as e:
+            self.module.fail_json(msg="Error getting distribution id from domain name = " + str(e), exception=traceback.format_exc(e))
+
+    def paginated_response(self, func, result_key, next_token=None):
+        '''
+        Returns expanded response for paginated operations.
+        The 'result_key' is used to define the concatenated results that are combined from each paginated response.
+        '''
+        args=dict()
+        if next_token:
+            args['NextToken'] = next_token
+        response = func(**args)
+        result = response.get(result_key)
+        next_token = response.get('NextToken')
+        if not next_token:
+           return result
+        return result + self.paginated_response(func, result_key, next_token)
+
+def to_dict(items, key, value):
+    ''' Transforms a list of items to a Key/Value dictionary '''
+    if items:
+        return dict(zip([i[key] for i in items], [i[value] for i in items]))
```

Comment:
If you allow me a suggestion here:

```python
def _extract_key_from_object_list(items, key, value):
    return dict((item[key], item[value]) for item in items)
```

You do not need to add the if-else construct, unless of course `items` may not be iterable.

Also, I don't see any usage of this helper function in this module, perhaps it is not needed after all?